### DISCUSSION TITLE
torch-related error when installing whisper on raspberry pi 4

### DISCUSSION THREAD

**Discussion by StewRat83** [openai/whisper#210] [Q&A] [answered]
Hi

Trying to install whisper on a raspberry pi 4.

There's another post that is getting the same error on Windows and the answer was about lack of pre-build PyTorch packages for their Windows/Linux environment, but I have been following a couple of raspi-specific posts to do this, and neither of them hits this problem it seems.
Relatively new to much of this, so suggestions welcome :)

There's another post 

$ pip install git+https://github.com/openai/whisper.git
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting git+https://github.com/openai/whisper.git
  Cloning https://github.com/openai/whisper.git to /tmp/pip-req-build-y8iiti6v
  Running command git clone -q https://github.com/openai/whisper.git /tmp/pip-req-build-y8iiti6v
Requirement already satisfied: numpy in /usr/lib/python3/dist-packages (from whisper==1.0) (1.19.5)
ERROR: Could not find a version that satisfies the requirement torch (from whisper)
ERROR: No matching distribution found for torch

**Comment by jongwook** [comment, marked as answer]
From [this article](https://pytorch.org/tutorials/intermediate/realtime_rpi.html), it seems there is an aarch64 build of PyTorch that runs on Raspberry Pi. Which OS / python version are you running?

**Reply by StewRat83** [reply]
Thanks.
I'm using the Raspbian v11 (bullseye) that came with the Pi, and Python 3.9.2.

**Reply by StewRat83** [reply]
Been doing some more reading and should clarify it is the 32-bit OS I am running, so sounds like I should set up a new SD card with the 64-bit OS and try again.

**Reply by StewRat83** [reply]
Closing the loop on this, I created an SD with the 64 bit PiOS, and whisper installs and runs fine.
There's a bit of a hiccup with the numpy dependency (maybe only if you don't already have that installed) as it seems to need a separate upgrade after the install, but that is well enough documented elsewhere.
Thanks
